### PR TITLE
[DUOS-2559] Request access link

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2559

### Summary

- Add checkmarks to the table so that you can multiselect datasets
- Add a link to the table to request access, creates a draft (call endpoint with correct parameters) and redirect to DAR edit view

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/CollapsibleTable.js [hunk 1]
```diff
@@ -166,50 +145,101 @@ const CollapsibleRow = (props) => {
...
-      return;
+    const isSelected = (id) => selected.indexOf(id) !== -1;
+    if (!isEmpty(data)) {
+      setAllSelected(data.rows.every((row) => {
+        return row.subtable.rows.every((subRow) => isSelected(subRow.id));
+      }));
+      setSomeSelected(data.rows.some((row) => {
+        return row.subtable.rows.some((subRow) => isSelected(subRow.id));
+      }));
     }
```
Comment (rushtong) [inline src/components/CollapsibleTable.js:211]: This is tricky to get right, nice work.

File: src/components/data_search/DatasetSearchTable.js [hunk 2]
```diff
@@ -28,8 +29,42 @@ const datasetTableHeader = [
...
+    } else {
+      newSelected = selected.concat(idsToModify);
+    }
+
+    setSelected(newSelected);
+  };
+
+  const applyForAccess = async () => {
+    const draftDatasets = selected.map((id) => parseInt(id.replace('dataset-', '')));
+    const darDraft = await DAR.postDarDraft({ datasetId: draftDatasets });
```
Comment (rushtong) [inline src/components/data_search/DatasetSearchTable.js:65]: Not your code, but I'm confused about why the singular `datasetId` is used for a list of ids. Probably a legacy code issue. Maybe we can clean this up in a different ticket.

**Review by rushtong — APPROVED** [review]
Looks great, works great!

**Review by hams7504 — APPROVED** [review]

File: src/components/data_search/DatasetSearchTable.js (hunk 2 above)
Comment (fboulnois) [reply src/components/data_search/DatasetSearchTable.js:65]: I created a ticket for this: https://broadworkbench.atlassian.net/browse/DUOS-2652

**Pull request merged**